### PR TITLE
Add Fost-Request-ID in every requests and responses

### PR DESCRIPTION
**Pull request by tleekkul**

### CODE REVIEW

File: Cpp/fost-urlhandler/routing.cpp [hunk 1]
```diff
@@ -34,6 +35,13 @@ bool fostlib::urlhandler::service(fostlib::http::server::request &req) {
     // Now process it
     fostlib::string hostname(req.data()->headers()["Host"].value());
 
+    // Fost-Request-ID
+    if (!req.data()->headers().exists("Fost-Request-ID")) {
+        req.data()->headers().set(
+                "Fost-Request-ID",
+                fostlib::timestamp_nonce24b64u().underlying().underlying());
```
Comment (tleekkul) [inline Cpp/fost-urlhandler/routing.cpp]: @KayEss Is it worth if we make this request_id generation to be a lambda function so we can test it as well?

Comment (KayEss) [reply Cpp/fost-urlhandler/routing.cpp]: You want to be able to pass a lambda into the service function?

Comment (tleekkul) [reply Cpp/fost-urlhandler/routing.cpp]: I want to be able to override the request id generation, so I can write test and compare it. Not sure it worth the effort or not.

File: Cpp/fost-urlhandler/view.request-id.tests.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/*
+    Copyright 2018-2019 Felspar Co Ltd. http://odin.felspar.com/
```
Comment (KayEss) [inline Cpp/fost-urlhandler/view.request-id.tests.cpp]: Should just be 2019

**Pull request closed without merging**